### ISSUE TITLE
illumos/solaris/smartos display compacted hwaddrs

### ISSUE THREAD

**Issue by sjorge** [saltstack/salt#31369]
```
[root@test ~]# salt-call grains.get hwaddr_interfaces
```
```yaml
local:
    ----------
    net0:
        0:22:6:10:0:1
    net1:
        0:22:6:30:0:3
```

These should probably be padded with 0 if length of segment < 2

Not sure how far back this goes but at least develop and 2016.3 show it. I assume 2015.8 and older also.

**Comment by ssgward** [comment]
@sjorge - thanks.

**Comment by sjorge** [comment]
Trying to fix it now in 2016.3 so I can test, then I will backport to 2015.5 as the code is the same.
I'll open the PR against 2015.5. Not sure how long it will take me.

**Comment by sjorge** [comment]
```
[root@nacl /salt_dev/salt]# /opt/salt/bin/salt-call --local grains.get hwaddr_interfaces
[INFO    ] Determining pillar cache
local:
    ----------
    net0:
        00:15:00:05:d2:e1
```

Fixed in 2016.3 on my dev box, expect PR soon

**Comment by ssgward** [comment]
@sjorge - good work. Thanks.

**Comment by sjorge** [comment]
Done PR #31374

**Comment by sjorge** [comment]
Oh nice this got merged into 2016.3!
